FEAT : ODYA-234 내 추억 및 여행일지 조회

leeJongMin58 opened the pull request:
### 개요
- 내 추억
- 여행일지 조회
- 아직 여행일지 수정은 X -> api 수정되면 정리해서 올리겠습니다.

### 변경사항
![KakaoTalk_Photo_2024-01-12-08-41-56 007](https://github.com/weIT-1st/Odya-android/assets/151716813/aee581bf-d200-4adc-b160-f1c3f903fe8c)
![KakaoTalk_Photo_2024-01-12-08-41-55 006](https://github.com/weIT-1st/Odya-android/assets/151716813/1e957708-f44b-491e-ba70-9796326326b3)
![KakaoTalk_Photo_2024-01-12-08-41-55 005](https://github.com/weIT-1st/Odya-android/assets/151716813/9d76e443-41e7-4100-8df0-5fa53a74620e)
![KakaoTalk_Photo_2024-01-12-08-41-55 004](https://github.com/weIT-1st/Odya-android/assets/151716813/bd23c320-4e8f-4394-a84d-a5cab9bc21ad)
![KakaoTalk_Photo_2024-01-12-08-41-55 003](https://github.com/weIT-1st/Odya-android/assets/151716813/e8cc00ee-1815-4fa3-bcf8-114c235497f7)
![KakaoTalk_Photo_2024-01-12-08-41-55 002](https://github.com/weIT-1st/Odya-android/assets/151716813/4b152251-0cf8-4c97-9de9-c61ca65a4cd0)
![KakaoTalk_Photo_2024-01-12-08-41-54 001](https://github.com/weIT-1st/Odya-android/assets/151716813/75e7a59c-fac7-456d-89c0-1ba6ed688144)

### 관련 지라 및 위키 링크
https://weit.atlassian.net/browse/ODYA-234?atlOrigin=eyJpIjoiMjRiZmZmZDAxZGE2NGUxOTljYzQxM2NhMzRhYzgzY2UiLCJwIjoiaiJ9

### 리뷰어에게 하고 싶은 말
감사합니다.

Commit:
FEAT : memory

Commit:
DESIGN : 내 여행일지 레이아웃 완성

Commit:
FEAT : 내 여행일지 (리사이클러뷰) 적용 커밋

-> 스크롤 문제 해결 필요

Commit:
FEAT : 임시 커밋

Commit:
feat : temp commit

Commit:
chore : api 수정 전 임시 커밋

Commit:
feat : 나의 여행일지 조회 및 내 추억

여행일지 수정은 api 나오면 추가 하겠습니다.

overthename commented on data/src/main/java/com/weit/data/service/BookMarkService.kt:
```diff
@@ -23,6 +23,14 @@ interface BookMarkService {
 
+    @GET("/api/v1/travel-journal-bookmarks/{userId}")
+    suspend fun getUserJournalBookMark(
+        @Path("userID") userId: Long,
```
userId 오타!

overthename commented on presentation/src/main/java/com/weit/presentation/model/journal/TravelJournalUpdateDTO.kt:
```diff
@@ -0,0 +1,14 @@
+    val travelStartDate: LocalDate,
+    val travelEndDate: LocalDate,
+    val visibility: String?,
+//    val travelJournalCompanions: List<TravelJournalCompanionsInfo>,
```
엇 여기는 왜 주석처리하셨나용

leeJongMin58 replied:
이거는 기록해 두고 여행일지 수정 작업할떄 마무리 하겠습니다

minseonglove replied:
남겨둬야하는 주석이 있으면 이유도 같이 달아주면 좋아요. 다른 사람이 필요없는 코든줄 알고 없애버릴 수 있기 때문이에요

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/album/TravelJournalAlbumFragment.kt:
```diff
@@ -0,0 +1,56 @@
+
+    override fun onResume() {
+        super.onResume()
+        binding.root.requestLayout()
```
onResume() 에서 이걸 쓴 이유가 잇나요?

leeJongMin58 replied:
여행일지 조회할떄 앨범형이랑 기본형 뷰가 각각 뷰의 높이가 다른데 탭으로 이동할때 마다 그 뷰를 새로 계산해 주라고 적용 시켰습니다.

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/album/TravelJournalAlbumViewModel.kt:
```diff
@@ -0,0 +1,52 @@
+        viewModelScope.launch {
+            val journal = journalInfo.value
+            _journalContents.emit(journal.travelJournalContents)
+        }
```
fragment에서 travelJournalInfo를 viewmodel에 주입시키고 다시 frgament로 보내는데 fragment에서 바로 adapter에 데이터를 넣어주는 방법도 있지 않을까 viewmodel을 꼭써야하나..??

leeJongMin58 replied:
그래도 데이터는 뷰모델에 있어야 하지 않을까해서 이렇게 했는데... 지금은 가공은 안하지만 만약 가공이 필요하다면 이렇게하는 로직이 맞을거 같아서

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/basic/TravelJournalBasicViewModel.kt:
```diff
@@ -0,0 +1,50 @@
+    init {
+        getContentsInfo()
+        Log.d("jomi", "journal : $travelJournalInfo")
+        Log.d("jomi", "contents : ${travelJournalInfo.travelJournalContents}")
```
주석을 안지우셨군요ㅎ

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/friends/TravelJournalFriendsFragment.kt:
```diff
@@ -0,0 +1,55 @@
+        binding.rvTravelJournalFriends.adapter = travelJournalFriendAdapter
+
+        travelJournalFriendAdapter.submitList(viewModel.friends)
+    }
```
viewmodel데이터를 가져다 쓰는건 안되는거라고 배웠는데.. 매개변수로 받은 friends를 바로 넣어도 될 것 같아욤

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/MemoryFragment.kt:
```diff
@@ -0,0 +1,223 @@
+        binding.rvJournalMemoryBookmarkJournal.adapter = null
+        binding.rvJournalMemoryTagJournal.adapter = null
+        binding.rvJournalMemoryMyReview.adapter = null
+        super.onDestroyView()
```
removeOnScrollListener() 도 해주세요!

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/MemoryFragment.kt:
```diff
@@ -0,0 +1,223 @@
+            adapter = myReviewAdapter
+        }
+
+    }
```
initRV 는 하나의 함수 안에서 해줘도 될 것 같네요 굳이 안바꿔되지만..

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/adapter/TaggedJournalAdapter.kt:
```diff
@@ -0,0 +1,96 @@
+
+            binding.toggleItemJournalMemoryBookmark.setOnClickListener {
+                // todo bookmark
+            }
```
엇 이건 어떤 기능이 남은거죠? 중복코드라 지워도 되는건가?

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/MemoryReviewViewModel.kt:
```diff
@@ -0,0 +1,84 @@
+            val result = deletePlaceReviewUseCase(placeReviewId)
+
+            if (result.isSuccess){
+                Log.d("memory", "Delete Review Success")
```
저희의 영원한 숙명..

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/MemoryReviewViewModel.kt:
```diff
@@ -0,0 +1,84 @@
+
+                if (newReviews.isEmpty()) {
+                    loadNextReviews()
+                }
```
이 if문 꼭 필요한가?? 나도 지웠거든

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/MyJournalViewModel.kt:
```diff
@@ -0,0 +1,185 @@
+                if (journals.isEmpty()){
+                    _isEmptyMyJournal.emit(true)
+                } else {
+                    _isEmptyMyJournal.emit(false)
```
여긴 boolean값을 조건문으로 대체하여 한줄로도 작성할 수 있을 것 같습니다

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/MyJournalViewModel.kt:
```diff
@@ -0,0 +1,185 @@
+                return@launch
+            }
+
+            if (journal.isBookmarked) {
```
밑에 두 함수가 같은내용이 중복되니 val result = if(journal.isBookmarked) createJournalBookMarkUseCase(journal.travelJournalId)이런 식으로 하나로 통합할 수 잇을 것 같아요

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/OtherJournalViewModel.kt:
```diff
@@ -0,0 +1,228 @@
+//                Log.d("memory", "Get Tagged Journal Fail : ${result.exceptionOrNull()}")
+//            }
+//        }
+//    }
```
여긴 주석처리한이유가 있나??

leeJongMin58 replied:
삭제

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/OtherJournalViewModel.kt:
```diff
@@ -0,0 +1,228 @@
+            }
+
+            if (journal.isBookmarked) {
+                deleteTravelJournalBookmark(journals, journal)
```
헷갈리니까 함수명을 tagged들어간걸로 바꾸면 더 좋을 것 같가유 createBookmarkTravelJournalBookmark이 함수명도 bookmark가 두번들어가서 가독성이 좋지 않네용

leeJongMin58 replied:
다시 보니까 중복이네요,,,

overthename commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/OtherJournalViewModel.kt:
```diff
@@ -0,0 +1,228 @@
+
+
+    fun deleteTaggedJournal() {
+        // todo 태그된 여행일지 삭제
```
이건 완료된거지?

leeJongMin58 replied:
요것도 수정 파트에서 마무리 될 예정입니다.

overthename reviewed (commented):
많은 작업량하신다고 고생하셨습니다~

Commit:
feat : 수정 ver1

Commit:
feat : 수정 ver1

Commit:
feat : 무한스크롤 수정 + 랜덤 여행일지 수정

Commit:
feat : 메모리 뷰 수정 및 무한스크롤 정리

Commit:
feat : 리뷰 수정

1. 여행일지 디테일 화살표 방향 수정
2. 내추억 중첩된 느낌의 뷰 dp 수정

Commit:
Merge branch 'develop' into feature/ODYA-234

# Conflicts:
#	data/src/main/java/com/weit/data/model/journal/TravelJournalListDTO.kt
#	data/src/main/java/com/weit/data/repository/journal/TravelJournalRepositoryImpl.kt
#	domain/src/main/java/com/weit/domain/model/journal/TravelJournalListInfo.kt
#	presentation/src/main/java/com/weit/presentation/ui/searchplace/SearchPlaceBottomSheetFragment.kt
#	presentation/src/main/res/layout/bottom_sheet_feed_my_menu.xml
#	presentation/src/main/res/layout/bottom_sheet_feed_other_menu.xml
#	presentation/src/main/res/layout/fragment_login_input_user_info.xml
#	presentation/src/main/res/layout/fragment_memory.xml
#	presentation/src/main/res/navigation/main_graph.xml
#	presentation/src/main/res/values/styles.xml

Commit:
feat : 충돌 해결

minseonglove commented on data/src/main/java/com/weit/data/repository/bookmark/BookMarkRepositoryImpl.kt:
```diff
@@ -65,6 +66,48 @@ class BookMarkRepositoryImpl @Inject constructor(
+            val userJournal = result.getOrThrow()
+            hasNextUserBookMark.set(userJournal.hasNext)
+            Result.success(userJournal.content.map{
+                JournalBookMarkInfo(
+                    it.travelJournalBookMarkId,
+                    it.travelJournalId,
+                    it.title,
+                    it.travelStartDate,
+                    it.travelJournalMainImageUrl,
+                    Writer(
+                        it.writer.userId,
+                        it.writer.nickname,
+                        it.writer.profile,
+                        it.writer.isFollowing
+                    )
+                )
```
요런 단순 캐스팅 동작은 확장함수로 빼면 코드가 더 이뻐짐니다

leeJongMin58 replied:
지혜가 250 브런치에서 수정을 한걸로 알고 있기 때문에 지금은 손대지 않을게요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/album/TravelJournalAlbumAdapter.kt:
```diff
@@ -0,0 +1,69 @@
+            binding.lyItemJournalDetailAlbum.isGone = isChecked
+            }
+
+            binding.btnItemJournalDetailKebab.setOnClickListener {
+                // todo 여행일지 컨텐츠 수정
+            }
```
TODO 표시 좋습니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/album/TravelJournalAlbumAdapter.kt:
```diff
@@ -0,0 +1,69 @@
+            binding.tvItemJournalDetailAlbumDate.text = item.travelDate
+            binding.tvItemJournalDetailAlbumContent.text = item.content
+            binding.tvItemJournalDetailAlbumPlace.text = item.placeDetail.name
+            binding.tvItemJournalDetailAlbumAddress.text =
+                item.placeDetail.address?.split(" ")?.slice(1..2)?.joinToString(" ") ?: ""
+
```
왠만하면 데이터의 가공은 ViewModel에서 한다음 넘겨주는게 베스트에요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/album/TravelJournalAlbumFragment.kt:
```diff
@@ -0,0 +1,56 @@
+        }
+    }
+
+    private fun initTravelJournalAlbumRV() {
+        binding.rvItemJournalDetailModel.apply {
+            adapter = travelJournalAlbumAdapter
+            addItemDecoration(SpaceDecoration(resources, bottomDP = R.dimen.item_memory_day_space))
+        }
+    }
```
adapter를 넣었다면 destory에서 없애줍시당

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/basic/TravelJournalBasicAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+import com.weit.presentation.databinding.ItemJournalDetailBasicModelBinding
+import com.weit.presentation.ui.util.SpaceDecoration
+
+class TravelJournalBasicAdapter: ListAdapter<TravelJournalContentsInfo, TravelJournalBasicAdapter.ViewHolder>(diffUtil) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
```
뭔가 위에 되게 비슷한 Adapter가 있던걸로 기억하는데 공통 파일로 묶을순 없었나요?

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/basic/TravelJournalBasicFragment.kt:
```diff
@@ -0,0 +1,56 @@
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class TravelJournalBasicFragment(
+    private val travelJournalInfo: TravelJournalInfo
+): BaseFragment<ItemJournalDetialModelBinding>(
+    ItemJournalDetialModelBinding::inflate
+) {
```
이것도 마찬가지 공통으로 묶을 수 없다면 이유도 함께 적어주세요

leeJongMin58 replied:
이거 공통된 부분 상위 클래스로 만들어서 상속 받는거 제가 못하는건지 안되는 건지 잘 모르겠습니다.. 한번 같이 봐주세요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/MemoryFragment.kt:
```diff
@@ -0,0 +1,228 @@
+    private val otherJournalViewModel: OtherJournalViewModel by viewModels()
+    private val reviewViewModel: MemoryReviewViewModel by viewModels()
+
+    private val myJournalAdapter = MyJournalAdapter(
+        { myJournalViewModel.moveToJournal(it)},
+        { myJournalViewModel.updateTravelJournalBookmarkState(it) }
+        )
+    private val bookmarkJournalAdapter = BookmarkJournalAdapter (
+        { moveToJournalDetail(it) },
+        { otherJournalViewModel.updateBookmarkTravelJournalBookmarkState(it) }
+    )
+    private val taggedJournalAdapter = TaggedJournalAdapter(
+        { moveToJournalDetail(it) },
+        { otherJournalViewModel.updateTaggedTravelJournalBookmarkState(it) },
+        { otherJournalViewModel.deleteTaggedJournal() }
+    )
```
각 콜백의 이름도 같이 적어주면 코드 이해에 조습니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/MemoryFragment.kt:
```diff
@@ -0,0 +1,228 @@
+
+        repeatOnStarted(viewLifecycleOwner){
+            myJournalViewModel.randomJournal.collectLatest {
+                if (it != null) {
+                    Glide.with(requireContext())
+                        .load(it.contentImageUrl)
+                        .into(binding.includeJournalMemoryLastJournal.ivItemJournalMemoryLastJournal)
```
null만 들어온 상황이라면 아예 이미지가 안뜨는일이 생길텐데 이 부분은 괜찮나요?

leeJongMin58 replied:
넵 null 일때는 해당 뷰가 gone으로 바뀌기 떄문에 괜찮습니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/adapter/TaggedJournalAdapter.kt:
```diff
@@ -0,0 +1,96 @@
+        PopupMenu(contextThemeWrapper, it).apply {
+            menuInflater.inflate(R.menu.menu_memory, this.menu)
+
+            setOnMenuItemClickListener {
+                when (it.itemId) {
+                    R.id.item_memory_delete -> {
+                        deleteItem
+                    }
+                }
+                false
+            }
```
혹씨 true false의 차이가 뭔지 알구잇나요

leeJongMin58 replied:
폴스일떄 이벤트가 계속 유지되는 걸로 알고 있습니다.

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/MemoryReviewViewModel.kt:
```diff
@@ -0,0 +1,80 @@
+
+                val originalReviews = reviews
+                reviews.clear()
+                reviews.addAll(originalReviews + newReviews)
```
originalReviews가 reviews의 참조를 가져오는거라면 reviews가 clear되는 시점에서 originalReviews도 없어질거 같은데 요 부분은 괜찮나요?

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/viewmodel/OtherJournalViewModel.kt:
```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    fun updateBookmarkTravelJournalBookmarkState(travelJournalId: Long) {
+        viewModelScope.launch {
+            val journals = bookMarkTravelJournals.value
+            val journal = journals.find { it.travelJournalId == travelJournalId }
```
아래 메소드와 상당히 유사하네요 합칠순없나요?

leeJongMin58 replied:
변수가 너무 많은거 같아서 구별하기 쉽게 뷰모델을 나눠 봤는데 하나로 합치는게 더 좋을까요??

minseonglove replied:
네

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/travel_journal/TravelJournalFragment.kt:
```diff
@@ -0,0 +1,265 @@
+                    val latitudes = emptyList<Double>()
+                    val longitudes = emptyList<Double>()
+                    info.travelJournalContents.forEach {
+                        val lat = it.placeDetail.latitude
+                        val lng = it.placeDetail.longitude
+
+                        if (lat == null) {
+                            return@forEach
+                        }
+
+                        if (lng == null) {
+                            return@forEach
+                        }
+
+                        latitudes.plus(lat)
+                        longitudes.plus(lng)
+                        val marker = MarkerOptions().apply {
+                            position(LatLng(lat, lng))
+                            icon(marker)
+                        }
+                        map?.addMarker(marker)
```
map에 마커를 추가하는 동작은 메소드로 빼도 좋겠어요 지금은 collect안에 일어나는 동작이 많이 기네요

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/travel_journal/TravelJournalViewModel.kt:
```diff
@@ -0,0 +1,162 @@
+        getJournalInfo()
+    }
+
+    private fun getJournalInfo() {
```
get은 뭔갈 리턴한다는 너낌이 강한 prefix에요 초기화에만 사용하기 때문에 initJournalInfo도 좋겠습니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/travel_journal/TravelJournalViewModel.kt:
```diff
@@ -0,0 +1,162 @@
+        val info = journalInfo.value ?: return emptyList()
+
+        val friendCount = info.travelJournalCompanions.size
+        return if (friendCount < MAX_ABLE_SHOW_FRIENDS_NUM) {
+            info.travelJournalCompanions
+        } else {
+            info.travelJournalCompanions.slice(0 until MAX_ABLE_SHOW_FRIENDS_NUM)
+        }
```
요것도 저번처럼 min을 활용해주십시오

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/update/journal/TravelJournalUpdateFragment.kt:
```diff
@@ -0,0 +1,100 @@
+            }
+        }
+
+//        repeatOnStarted(viewLifecycleOwner) {
+//            viewModel.travelFriendsInfo.collectLatest { info ->
+//                travelFriendsAdapter.submitList(info.friendsSummary)
+//                if (info.remainingFriendsCount > 0) {
+//                    binding.includePostTravelLogFriends.tvTravelFriendsCount.text =
+//                        getString(R.string.post_travel_log_friends_count, info.remainingFriendsCount)
+//                }
+//            }
+//        }
```
모야

leeJongMin58 replied:
작업중에 아직 api가 수정중이라고 해서 일단 정지한 상태 입니다. todo로 남겨두겠습니다

minseonglove reviewed (commented):
일단 여까지

Commit:
feat : 리뷰 수정 2차

1. todo를 이용해서 추가 작업할 내용 표시
2. AlbumAdater/BasicAdater에서 하던 데이터 가공을 뷰모델로 이동
3. destroy시 adapter 삭제
4. 콜백 이름 적어주기
5. 무한스크롤시 clear() 문제 해결
6. get과 init 접두사로 역할 분리
7. travelJournalMap에 마커 추가를 메서도로 분리

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/journal/memory/MemoryFragment.kt:
```diff
@@ -35,16 +35,16 @@ class MemoryFragment : BaseFragment<FragmentMemoryBinding>(
-        { moveToJournalDetail(it) },
-        { otherJournalViewModel.updateTaggedTravelJournalBookmarkState(it) },
+        { journalId -> moveToJournalDetail(journalId) },
+        { journalId -> otherJournalViewModel.updateTaggedTravelJournalBookmarkState(journalId) },
```
콜백안의 매개변수가 아이고 콜백자체 이름을 명시하면 가독성이 좋아진다는 것이였어요

minseonglove reviewed (approved):
* 지혜도 그렇고 다른 브랜치에서 수정했기 때문에 냅둔다는 리뷰가 종종 보입니다. 하지만 처음부터 머지안된 코드를 끌어쓰는것은 지양해야해요. 코드 리뷰를 거치면서 수정될 위험이 있고 끌어쓰는 사람이 코드에서 문제를 발견해도 수정하기가 힘들기 때문이에요 (충돌문제)

* 공통 로직을 상위 클래스로 묶는게 어렵다고 했는데 Fragment는 binding객체 자체가 달라져서 힘들 순 있지만 그외 ViewModel 같은건 충분히 공통 로직을 뺄 수 있다고 봅니다. 여기서 공통 로직의 기준은 어떤 파일의 로직이 수정됐을때 그 파일과 유사한 다른 파일도 로직을 수정해야할 정도로 둘이 똑같이 움직이는 로직을 말해요.

Commit:
Merge branch 'develop' into feature/ODYA-234

# Conflicts:
#	presentation/src/main/res/layout/item_friend_manage_follow.xml
